velox/functions/prestosql/tests/ArrayHasDuplicatesTest.cpp

```diff
@@ -0,0 +1,147 @@
+      {std::nullopt},
+      {1, 2, 3},
+      {2, 1, 1, -2},
+      {1, 1, 1},
+      {-1, std::nullopt, -1, -1},
+      {std::nullopt, std::nullopt, std::nullopt},
+      {1, -2, -2, 8, -2, 4, 8, 1},
+      {std::numeric_limits<int64_t>::max(),
+       std::numeric_limits<int64_t>::max(),
+       1,
+       std::nullopt,
+       0,
+       1,
+       std::nullopt,
+       0},
+  });
+
+  auto expected = makeNullableFlatVector<bool>(
+      {false, false, false, false, true, true, true, true, true, true});
+
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
+
+// Test inline (short) strings.
+TEST_F(ArrayHasDuplicatesTest, inlineStrings) {
+  using S = StringView;
+
+  auto array = makeNullableArrayVector<StringView>({
+      {},
+      {""_sv},
+      {std::nullopt},
+      {S("a"), S("b")},
+      {S("a"), std::nullopt, S("b")},
+      {S("a"), S("a")},
+      {S("b"), S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, std::nullopt},
+      {S("b"), std::nullopt, S("a"), S("a"), std::nullopt, S("b")},
+  });
+
+  auto expected = makeFlatVector<bool>(
+      {false, false, false, false, false, true, true, true, true});
+
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
+
+// Test non-inline (> 12 character length) strings.
+TEST_F(ArrayHasDuplicatesTest, longStrings) {
+  using S = StringView;
+
+  auto array = makeNullableArrayVector<StringView>({
+      {S("red shiny car ahead"), S("blue clear sky above")},
+      {S("blue clear sky above"),
+       S("yellow rose flowers"),
+       std::nullopt,
+       S("blue clear sky above"),
+       S("orange beautiful sunset")},
+      {
+          S("red shiny car ahead"),
+          std::nullopt,
+          S("purple is an elegant color"),
+          S("red shiny car ahead"),
+          S("green plants make us happy"),
+          S("purple is an elegant color"),
+          std::nullopt,
+          S("purple is an elegant color"),
+      },
+  });
+  auto expected = makeFlatVector<bool>({false, true, true});
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
+
+TEST_F(ArrayHasDuplicatesTest, nullFreeBigints) {
+  auto array = makeArrayVector<int64_t>({
+      {1,
+       std::numeric_limits<int64_t>::min(),
+       std::numeric_limits<int64_t>::max()},
+      {2, 1, 1, -2},
+      {1, 1, 1},
+  });
+
+  auto expected = makeNullableFlatVector<bool>({false, true, true});
+
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
+
+TEST_F(ArrayHasDuplicatesTest, nullFreeStrings) {
+  using S = StringView;
+
+  auto array = makeArrayVector<StringView>(
+      {{S("red shiny car ahead"), S("blue clear sky above")},
+       {S("red shiny car ahead"),
+        S("blue clear sky above"),
+        S("blue clear sky above")},
+       {S("a"), S("b")},
+       {S("a"), S("b"), S("b")}
+
+      });
+  auto expected = makeFlatVector<bool>({false, true, false, true});
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
```

Comment:
Thanks for the contribution, @duanmeng. I have closed my duplicate PR. Feel free to borrow more unit test cases, such as constant vectors, nonContiguousRows, here.

https://github.com/facebookincubator/velox/pull/3397/files#diff-198e97a86767f3f7b75862d0350895b93496dc376dbf8a1fe3bc384cec368076

Reply:
> Thanks for the contribution, @duanmeng. I have closed my duplicate PR. Feel free to borrow more unit test cases, such as constant vectors, nonContiguousRows, here.
> 
> https://github.com/facebookincubator/velox/pull/3397/files#diff-198e97a86767f3f7b75862d0350895b93496dc376dbf8a1fe3bc384cec368076

@darrenfu Thanks for your suggestion, I used to add the constant and non Contiguous Rows test case referring to the test case in `Array Duplicates Test` and removed them according to the comments *"This test is redundant as this logic is processed by the framework"*.